integration_test/playertesting/expectations/MultipleEventsExpectation.ts

```diff
@@ -0,0 +1,155 @@
+        });
+      })
+      .join(' - ');
+  }
+}
+
+// To expect multiple events in any order.
+export class EventBagExpectation extends MultipleEventsExpectation {
+  readonly singleExpectations: SingleEventExpectation[];
+
+  constructor(
+    singleExpectationConvertibles: SingleEventExpectation[] | EventType[]
+  ) {
+    super();
+    this.singleExpectations = singleExpectationConvertibles.map(
+      (singleExpectationConvertible) =>
+        singleExpectationConvertible instanceof SingleEventExpectation
+          ? singleExpectationConvertible
+          : new PlainEventExpectation(singleExpectationConvertible as EventType)
+    );
+  }
+
+  get expectedFulfillmentCount(): number {
+    return this.singleExpectations.length;
+  }
+
+  isNextExpectationMet(receivedEvent: Event): boolean {
+    const nextUnfulfilled = this.singleExpectations.find(
+      (expectation) =>
+        !expectation.isFulfilled &&
+        expectation.maybeFulfillExpectation(receivedEvent)
+    );
+
+    return nextUnfulfilled?.isFulfilled ?? false;
+  }
+
+  toString(): string {
+    return this.singleExpectations
+      .map((expectation) => expectation.toString())
+      .join(' - ');
+  }
+}
+
+// To expect a given eventType to happen multiple times.
+export class RepeatedEventExpectation extends EventSequenceExpectation {
+  constructor(
+    singleExpectationConvertible: SingleEventExpectation | EventType,
+    count: number
+  ) {
+    const singleExpectation =
+      singleExpectationConvertible instanceof SingleEventExpectation
+        ? singleExpectationConvertible
+        : new PlainEventExpectation(singleExpectationConvertible as EventType);
+    super(Array.from({ length: count }, () => singleExpectation.copy()));
+  }
+}
+
+// To expect at least one out of multiple events.
+export class AnyEventExpectation extends MultipleEventsExpectation {
+  readonly singleExpectations: SingleEventExpectation[];
+
+  constructor(
+    singleExpectationConvertibles: SingleEventExpectation[] | EventType[]
+  ) {
+    super();
+    this.singleExpectations = singleExpectationConvertibles.map(
+      (singleExpectationConvertible) =>
+        singleExpectationConvertible instanceof SingleEventExpectation
+          ? singleExpectationConvertible
+          : new PlainEventExpectation(singleExpectationConvertible as EventType)
+    );
+  }
+
+  get expectedFulfillmentCount(): number {
+    return 1;
+  }
+
+  isNextExpectationMet(receivedEvent: Event): boolean {
+    const nextUnfulfilled = this.singleExpectations.find((expectation) =>
+      expectation.maybeFulfillExpectation(receivedEvent)
+    );
+
+    return nextUnfulfilled?.isFulfilled ?? false;
+  }
+
+  toString(): string {
+    return this.singleExpectations
+      .map((expectation) => expectation.toString())
+      .join(' - ');
+  }
+}
+
+export class S extends EventSequenceExpectation {}
+export class B extends EventBagExpectation {}
+export class R extends RepeatedEventExpectation {}
```

Comment:
```suggestion
export function RepeatedEvent(
  singleExpectationConvertible: SingleEventExpectation | EventType,
  count: number
): EventSequenceExpectation {
  return new RepeatedEventExpectation(singleExpectationConvertible, count);
}
```

Reply:
added variadic parameters for easier usage in a4c62d7827661cd9260b462acdae7e9cbeb182bd